### DISCUSSION TITLE
AIR 33.1.1.856 IOS installation failed

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2013] [Q&A]
AIR 33.1.1.856 IOS installation failed
Strangely, some iPads can be installed, but some newly purchased iPads will prompt "cannot install this app because its integrity cannot be verified" when installing.

UDID confirms that it has been added to the new description file, but cannot be installed.

thank you!

**Comment by Dextercitox** [comment]
I had the same problem some weeks ago. In the Air SDK you have to open your adt.cfg file and uncomment ( or add ) the following line :
UseNativeCodesign=true

I also regenerated the certs and provision files, and that way those newer iPads and iPhones start installing again without issues.

Hope that it helps !

**Reply by ajwfrost** [reply]
Hmm, this sounds like Apple have (again) changed the code-signing requirements .. @Dextercitox I don't suppose you could send me the same application in two variants, where you have "true" and "false" for that "UseNativeCodesign" configuration? I would like to know what the differences are in the generated IPA files...

thanks

**Reply by Dextercitox** [reply]
I can send you both versions. Do you want them in your GitHub account email?

**Reply by ajwfrost** [reply]
Thanks! our email system probably won't cope with that, there's a link below that should allow you to upload them...
https://transfer.harman.com/requests/r2qiPxOk5FQKG2qMdPID76

thanks

**Reply by GeneralVimes** [reply]
> UseNativeCodesign=true

Just want to confirm - are you using MacOs for game building? Windows users who compile games for iOS have nothing to worry about for now, do they?

**Reply by elemdeso** [reply]
Just for info - still working on it - I have this same message when attempting to install an iOS .ipa on a Mac Studio (Monterey). And I have UseNativeCodesign set to TRUE. It's an AdHoc Distribution build with the studio Provisioning UDID included.

**Reply by ajwfrost** [reply]
@elemdeso what are you using to package the IPA with? is that macOS (and if so which OS version) or Windows?

**Reply by elemdeso** [reply]
@ajwfrost Thanks! I'm packaging with Animate 22.0.7 on MacOS 12.4 on an Intel Mac Mini. It's an up to date iPad-only iOS app - just a test to see how it scales on an ARM Mac. I've packaged it with a distribution certificate and an ad hoc distribution profile with the ARM Mac UDID included. I copy the .ipa to the ARM Mac and double click to install. Then I get the "cannot be verified" message. I'm using UseNativeCodesign=true.

**Reply by elemdeso** [reply]
Oh and using SDK 33.1.1.889

**Reply by ajwfrost** [reply]
@elemdeso okay thanks, I'm a little surprised/concerned that there's a verification error if you're using a mac with native codesigning and yet it's not properly verifying it. If you unzip the .ipa file on the mac (either one) and run `codesign -dvvv Payload/appname.app` in a terminal window where the ipa file is, do you see any error/warning messages? (or `codesign -vvvv Payload/appname.app`?).

I'm also wondering whether you could try packaging it manually rather than using Animate. I don't know of any reasons why Animate (on an Intel mac) would affect things, but it's always a possibility!

Also are you able to check with a really basic/empty application, no native extensions or anything? as there's always the possibility that it's missing a framework signature or similar. (Actually that makes me think .. we may have adjusted (corrected) the bundle format recently and potentially we're not then signing the right locations ... will double-check that!!)

thanks

**Reply by elemdeso** [reply]
This is great - much appreciated. I'll do this. My next step was to try a hello world app in any case. Though manual packaging - its been a long time. I've got some commands saved somewhere , I think. I don't believe the app that's failing includes any ANE.

**Reply by ajwfrost** [reply]
Thanks - I've just double-checked how we do a native code-sign and it shouldn't make a difference where the frameworks are, we search the whole package to find and sign them. So .. curious one.. we can also try to do a simple hello-world here to reproduce it.

**Reply by elemdeso** [reply]
Nothing suspicious yet. There were no errors/warnings in the unzipped .ipa. I also tried a hello world app with different publish settings. Nothing made any difference. I'm going to try adding the hardware UUID into the device list as well. Will then have a try at manual packaging.

**Reply by elemdeso** [reply]
@ajwfrost Problem solved. App now plays faultlessly on Mac. There was a problem with the certificate/profile. When Animate published, it gave me an .ipa but failed silently with no message. Manual packaging gave me the warning. Revoked and regenerated the cert and profile. Thanks again for your attention and care.

**Reply by ajwfrost** [reply]
Okay thanks for letting us know, I guess it would be nicer if we could warn about certificate/profile issues somehow.. :-)
thanks

**Comment by Jason0247** [comment]
If so, thank you very much! The problem has been solved! @Dextercitox

**Reply by ajwfrost** [reply]
@Jason0247 was that solved after you used the native codesign option? if so are you able to send us the two IPA files (i.e. the one that fails and the one that works) so that we can check the differences? the upload link is still valid: https://transfer.harman.com/requests/r2qiPxOk5FQKG2qMdPID76
Thanks

**Comment by GeneralVimes** [comment]
> some newly purchased iPads will prompt "cannot install this app because its integrity cannot be verified" when installing.

Does this happen when installing from the store?

**Comment by Jason0247** [comment]
Upload completed! @ajwfrost

**Reply by ajwfrost** [reply]
Thanks .. but those are two different applications? The non-working one is likely to be not working because it's built with an older AIR SDK. So it would be telling the device that it wasn't built with the latest SDK or Xcode, and the code signature has also been updated.

We could do with the same application, using the same AIR SDK version, where it's failing on some devices vs code-signed a different way and then working on all devices.

thanks

**Comment by Jason0247** [comment]
Yes, these are two different applications. The ones that are not easy to use are the p12 certificate and description file issued three months ago. Another one uses p12 files and description files that are re created from Apple's official website.
Because I didn't save two different files of the same project when republishing, I provided these two files. @ajwfrost
